test/hotspot/jtreg/serviceability/dcmd/compiler/PerfMapTest.java

```diff
@@ -96,13 +103,14 @@ public void specifiedMapFile() {
     }
```

Comment:
Q: It is not clear why the file is not deleted in this case. Do I miss anything?

Reply:
>  Q2: Can we use Files.deleteIfExists(path) in the finally block of the run() method instead?

I don't think that changes anything. If run() throws an exception and then Files.deleteIfExists() throws an exception from the finally block, the original exception is lost.

> One more approach is to wrap run() call into the runHelper() method:

Same issue with potentially losing the original exception.

Reply:
Two variants:
```
    public void runHelper(CommandExecutor executor, String cmd, Path path) {
       try {
           run(executor, cmd, path);
       } catch (Throwable t) {
            t.printStackTrace(System.out);
            throw t;
       } finally {
           Files.deleteIfExists(path);
       }
    }
```
or
```
    public void runHelper(CommandExecutor executor, String cmd, Path path) {
       try {
           run(executor, cmd, path);
       } finally {
           try {
              Files.deleteIfExists(path);
           } catch (Throwable t) {
              <print message about exception thrown from deleteIfExists>
              t.printStackTrace(System.out);
           }
       }
    }
```

Reply:
The one issue with the first variant is that when run() throws an exception, it will be printed twice, assuming deleteIfExists() does not throw an exception. Basically you either get the exception printed twice, or you get the exception printed once plus whatever exception deleteIfExists() throws also printed (which is ok).

I think your 2nd example will work, but it's kind of confusing because if you got into the finally block via an exception, you then might throw another exception, which is handled in the finally block, and then (implicitly) the original exception is rethrown as you exit the finally block. However, if there was no exception from run() but deleteIfExists() throws an exception, the exception is printed, but does not cause the test to fail. I don't think we want that.

Reply:
Hello Chris, I haven't followed this PR and this is just a drive by comment about the exception handling discussion. In some other areas of our JDK tests, for situations like this, we do the following:

```java
Throwable failure = null;
try {
    run(new JMXExecutor(), "Compiler.perfmap " + path.toString(), path);
} catch (Throwable t) {
    failure = t;
} finally {
    try {
        Files.deleteIfExists(path);
    } catch (IOException ioe) {
        if (failure != null) {
            // add the IOException as a suppressed exception to the original failure
            // and rethrow the original
            failure.addSuppressed(ioe);
            throw failure;
        }
        // no failures in the run(), but deleting the file failed, rethrow this IOException
        throw ioe;
    }
}
```
This way both the original test failure (if any) as well as the failure in the finally block (if any) are propagated and made available in the failure report.

Reply:
Hi Jai. I think that solves all the functional requirements, but now we've turned two easily understood lines of code (calling run() and deleteIfExists()) into what you have above. I'm not so sure it is worth it. I refer back to Leonid's comment above on this topic:

> I thought about execution of Files.deleteIfExists(path) in the case of test fails, but decided that it is not so important. Also, test always might fail with crash when no finally block is executed. So I am fine with current way.

Reply:
Hello Chris,
> Hi Jai. I think that solves all the functional requirements, but now we've turned two easily understood lines of code (calling run() and deleteIfExists()) into what you have above. 

I agree. 

> I'm not so sure it is worth it. I refer back to Leonid's comment above on this topic:
> 
> > I thought about execution of Files.deleteIfExists(path) in the case of test fails, but decided that it is not so important. Also, test always might fail with crash when no finally block is executed. So I am fine with current way.

What Leonid suggests looks fine to me and keeps things simple.

Reply:
Ok. I think then I'll just push this PR in its current form.